### PR TITLE
add SVCB based lookup to find DOH endpoints

### PR DESCRIPTION
**Pull request by cryptix**

### CODE REVIEW

File: golang/sig0/doh_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestFindDOHEndpoint(t *testing.T) {
+	endpoint, err := FindDOHEndpoint("zenr.io")
+	require.NoError(t, err)
+
+	want := "https://doh.zenr.io/dns-query"
+	require.Equal(t, want, endpoint.String())
+}
```
Comment (cryptix) [inline golang/sig0/doh_test.go:15]: if you know of other setups, the test case should be pretty easy to extend

File: golang/sig0/doh.go [hunk 2]
```diff
@@ -39,3 +46,79 @@ func SendUDPQuery(server string, m *dns.Msg) (*dns.Msg, error) {
...
+	if dohPath.Template == "" {
+		return nil, fmt.Errorf("findDOHEndpoint: expected dohPath Template value")
+	}
+
+	// TODO: write this more genericly but striclty speaky we should check for IN HTTPS to check if we can use https
+	var hasHttp = false
+	for _, a := range alpn.Alpn {
+		if strings.Contains(a, "h1") || strings.Contains(a, "h2") {
+			hasHttp = true
+		}
```
Comment (cryptix) [inline golang/sig0/doh.go:107]: doesn't check for http/3 yet but only supporting that seems a little fringe to me.

File: golang/sig0/doh.go [hunk 3]
```diff
@@ -39,3 +46,79 @@ func SendUDPQuery(server string, m *dns.Msg) (*dns.Msg, error) {
...
+		return nil, fmt.Errorf("findDOHEndpoint: expected alpn in svcb.Value")
+	}
+	if dohPath == nil {
+		return nil, fmt.Errorf("findDOHEndpoint: expected dohPath in svcb.Value")
+	}
+	if dohPath.Template == "" {
+		return nil, fmt.Errorf("findDOHEndpoint: expected dohPath Template value")
+	}
+
+	// TODO: write this more genericly but striclty speaky we should check for IN HTTPS to check if we can use https
```
Comment (cryptix) [inline golang/sig0/doh.go:102]: see chat

File: flake.lock
Comment (adam-burns) [inline flake.lock:1]: is flake.lock required?

Comment (cryptix) [reply flake.lock:1]: ah, no, not at all. I snug a nix-shell in as my development environment a while ago and just updated it.

File: golang/sig0/doh.go (hunk 2 above)
Comment (cryptix) [reply golang/sig0/doh.go:107]: but then again, i'm not even sure Go supports http/3 yet.

**Pull request merged**